interfaces/configfiles/backend_test.go

```diff
@@ -0,0 +1,297 @@
+	c.Assert(s.Repo.AddBackend(s.Backend), IsNil)
+}
+
+func (s *backendSuite) TearDownTest(c *C) {
+	dirs.SetRootDir("/")
+	s.restoreSanitize()
+}
+
+var _ = Suite(&backendSuite{})
+
+func (s *backendSuite) TestName(c *C) {
+	c.Check(s.Backend.Name(), Equals, interfaces.SecurityConfigfiles)
+}
+
+func (s *backendSuite) mockSlot(c *C, yaml string, slotName string) (*interfaces.SnapAppSet, *snap.SlotInfo) {
+	info := snaptest.MockInfo(c, yaml, nil)
+
+	set, err := interfaces.NewSnapAppSet(info, nil)
+	c.Assert(err, IsNil)
+	err = s.Repo.AddAppSet(set)
+	c.Assert(err, IsNil)
+
+	if slotInfo, ok := info.Slots[slotName]; ok {
+		return set, slotInfo
+	}
+	panic(fmt.Sprintf("cannot find slot %q in snap %q", slotName, info.InstanceName()))
+}
+
+func (s *backendSuite) mockPlugs(c *C, yaml string, plugNames []string) (*interfaces.SnapAppSet, []*snap.PlugInfo) {
+	info := snaptest.MockInfo(c, yaml, nil)
+
+	set, err := interfaces.NewSnapAppSet(info, nil)
+	c.Assert(err, IsNil)
+	err = s.Repo.AddAppSet(set)
+	c.Assert(err, IsNil)
+
+	plugInfos := make([]*snap.PlugInfo, 0, len(plugNames))
+	for _, plug := range plugNames {
+		if plugInfo, ok := info.Plugs[plug]; ok {
+			plugInfos = append(plugInfos, plugInfo)
+			continue
+		}
+		panic(fmt.Sprintf("cannot find plug %q in snap %q", plug, info.InstanceName()))
+	}
+	return set, plugInfos
+}
+
+const eglProvider1 = `name: egl1
+version: 0
+type: app
+slots:
+  egl-driver-libs:
+`
+
+const eglProvider2 = `name: egl2
+version: 0
+type: app
+slots:
+  egl-driver-libs:
+`
+
+const otherProvider = `name: other
+version: 0
+type: app
+slots:
+  other-driver-libs:
+`
+
+const eglConsumer = `name: snapd
+version: 0
+type: snapd
+apps:
+  app:
+    plugs: [egl-driver-libs, other-driver-libs]
+`
+
+func checkConfigfilesFile(c *C, path, content string) {
+	c.Assert(filepath.Join(dirs.GlobalRootDir, path), testutil.FileEquals, content)
+}
+
+func (s *backendSuite) TestSandboxFeatures(c *C) {
+	c.Assert(s.Backend.SandboxFeatures(), DeepEquals, []string{"mediated-configfiles"})
+}
+
+func (s *backendSuite) TestConnectDisconnect(c *C) {
+	// Add callback and register the interface
+	iface := &ifacetest.TestConfigFilesInterface{
+		TestInterface: ifacetest.TestInterface{
+			InterfaceName: "egl-driver-libs",
+			ConfigfilesConnectedPlugCallback: func(spec *configfiles.Specification,
+				plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+				switch slot.Snap().InstanceName() {
+				case "egl1":
+					spec.AddPathContent(
+						filepath.Join(dirs.GlobalRootDir, "/etc/conf1.d/a.conf"),
+						&osutil.MemoryFileState{Content: []byte("aaaa"), Mode: 0655})
+				case "egl2":
+					spec.AddPathContent(
+						filepath.Join(dirs.GlobalRootDir, "/etc/conf1.d/b.conf"),
+						&osutil.MemoryFileState{Content: []byte("bbbb"), Mode: 0655})
+				}
+				return nil
+			},
+		},
+		PathPatternsCallback: func() []string {
+			return []string{filepath.Join(dirs.GlobalRootDir, "/etc/conf1.d/*.conf")}
```

Comment:
could you use use some realistic paths in the examples/tests? we really don't want to remove paths that are not related to snapd/snaps so the pattern need to contain something about snaps/snapd

Reply:
Changed